packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -360,7 +366,11 @@ export function useConnections({
         Pick<ConnectionInfo, 'id'>
     ) => {
       try {
-        return await saveConnection(connectionInfo);
+        const isNewConnection = !getConnectionInfoById(connectionInfo.id);
+        const savedConnectionInfo = await saveConnection(connectionInfo);
+        if (isNewConnection) {
+          onConnectionCreated?.(savedConnectionInfo);
```

Comment:
Similar to other callbacks, can you wrap these in `useCurrentRef` and use refs instead to make sure that the functions you are passing these to are stable and don't have direct dependency on these callbacks?